spec/gutter-spec.coffee

```diff
@@ -0,0 +1,74 @@
+{Gutter, DEFAULT_PRIORITY} = require '../src/gutter'
+
+describe 'Gutter initialization tests.', ->
+  name = 'name'
+  fakeGutterContainer = {}
+
+  it 'Test initializing a Gutter with all options.', ->
+    location = 'left'
+    priority = 1
+    visibleInitially = false
+    options =
+      'location': 'left'
+      'priority': 1
+      'visibleInitially': false
+    gutter = new Gutter name, fakeGutterContainer, options
+    expect(gutter.name).toBe name
+    expect(gutter.gutterContainer).toBe fakeGutterContainer
+    expect(gutter.location).toBe location
+    expect(gutter.priority).toBe priority
+    expect(gutter.visible).toBe visibleInitially
+
+  it 'Test initializeing a Gutter with minimal options.', ->
+    fakeGutterContainer = {};
+    gutter = new Gutter name, fakeGutterContainer
+    expect(gutter.name).toBe name
+    expect(gutter.gutterContainer).toBe fakeGutterContainer
+    expect(gutter.location).toBe 'left' # default
+    expect(gutter.priority).toBe DEFAULT_PRIORITY
+    expect(gutter.visible).toBe true # default
+
+describe 'Test hiding and showing a gutter.', ->
+  fakeGutterContainer = {}
+  fakeListener =
+    onGutterShow: null
+    onGutterHide: null
+  name = 'name'
+
+  attachListenerToGutter = (gutter) ->
+    gutter.onShow () ->
+      fakeListener.onGutterShow()
+    gutter.onHide () ->
+      fakeListener.onGutterHide()
+    spyOn fakeListener, 'onGutterShow'
+    spyOn fakeListener, 'onGutterHide'
+
+  it 'Test hiding a visible Gutter.', ->
+    options = visibleInitially: true
+    gutter = new Gutter name, fakeGutterContainer, options
+    attachListenerToGutter(gutter)
+
+    expect(gutter.visible).toBe true
+    gutter.hide()
+    expect(gutter.visible).toBe false
+    expect(fakeListener.onGutterHide).toHaveBeenCalled()
+
+  it 'Test showing a hidden Gutter.', ->
```

Comment:
This test could be combined into test that hides the gutter when visible. It's subjective, but for simple things like that I often combine just to keep the test tighter.